pkg/cmd/certregenerationcontroller/cabundlesyncer.go

```diff
@@ -0,0 +1,148 @@
+		configMapLister: kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Lister(),
+		eventRecorder:   eventRecorder.WithComponentSuffix("manage-client-ca-bundle-recovery-controller"),
+		queue:           workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "CABundleRecoveryController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Informer().HasSynced)
+
+	handler := cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.queue.Add(workQueueKey) },
+		UpdateFunc: func(old, new interface{}) { c.queue.Add(workQueueKey) },
+		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
+	}
+
+	// we react to some config changes
+	namespaces := []string{
+		operatorclient.GlobalUserSpecifiedConfigNamespace,
+		operatorclient.GlobalMachineSpecifiedConfigNamespace,
+		operatorclient.OperatorNamespace,
+		operatorclient.TargetNamespace,
+	}
+	for _, namespace := range namespaces {
+		informers := kubeInformersForNamespaces.InformersFor(namespace)
+		informers.Core().V1().ConfigMaps().Informer().AddEventHandler(handler)
+		c.cachesToSync = append(c.cachesToSync, informers.Core().V1().ConfigMaps().Informer().HasSynced)
+	}
+
+	return c, nil
+}
+
+func (c *CABundleController) Run(ctx context.Context) {
+	defer utilruntime.HandleCrash()
+
+	// FIXME: These are missing a wait group to track goroutines and handle graceful termination
+	// (@deads2k wants time to think it through)
+
+	klog.Info("Starting CA bundle controller")
+	defer func() {
+		klog.Info("Shutting down CA bundle controller")
+		c.queue.ShutDown()
+		klog.Info("CA bundle controller shut down")
+	}()
+
+	if !cache.WaitForNamedCacheSync("CABundleController", ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	go func() {
+		wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}()
+
+	<-ctx.Done()
+}
+
+func (c *CABundleController) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *CABundleController) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.sync(ctx)
+
+	if err == nil {
+		c.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %w", key, err))
+	c.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (c *CABundleController) sync(ctx context.Context) error {
+	// Always start 10 seconds later after a change occurred. Makes us less likely to steal work and logs from the operator.
+	timer := time.NewTimer(10 * time.Second)
+	defer timer.Stop()
+	select {
+	case <-timer.C:
+	case <-ctx.Done():
+		return nil
+	}
+
+	_, changed, err := targetconfigcontroller.ManageClientCABundle(c.configMapLister, c.configMapGetter, c.eventRecorder)
+	if err != nil {
+		return err
+	}
+
+	if changed {
+		klog.Info("Refreshed client CA bundle.")
```

Comment:
this still needs to be fixed.  if it's top level info worth it should be an event. If it isn't (more likely) make it a V(2).

Reply:
sorry I missed adding v2 yesterday, will update it